src/test/test-util.c

```diff
@@ -86,24 +86,24 @@ static void test_max(void) {
         assert_se(LESS_BY(4, 8) == 0);
         assert_se(LESS_BY(16, LESS_BY(8, 4)) == 12);
         assert_se(LESS_BY(4, LESS_BY(8, 4)) == 0);
-        assert_se(CMP(-5, 5) == -1);
-        assert_se(CMP(5, -5) == 1);
+        assert_se(CMP(3, 5) == -1);
+        assert_se(CMP(5, 3) == 1);
         assert_se(CMP(5, 5) == 0);
         assert_se(CMP(x, y) == -1);
         assert_se(CMP(y, x) == 1);
         assert_se(CMP(x, x) == 0);
         assert_se(CMP(y, y) == 0);
-        assert_se(CMP(UINT64_MAX, 0L) == 1);
-        assert_se(CMP(0L, UINT64_MAX) == -1);
+        assert_se(CMP(UINT64_MAX, (uint64_t) 0) == 1);
+        assert_se(CMP((uint64_t) 0, UINT64_MAX) == -1);
```

Comment:
`0U` sounds shorter and nicer?

Reply:
It may be. But I am not sure...

Reply:
I think `0UL` would be correct (`unsigned long`, right?) though the unsignedness is the important part... Actually I think that's what I had in mind at first... Not sure why I didn't have the warnings myself, need to double check that. The cast is totally fine too, it's what I've done in cases tha broke for me. Thanks for fixing @yuwata and thanks for catching and reporting @evverx !